src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,46 @@
+package duke.command;
+
+import duke.Storage;
+import duke.TextUi;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * A FindCommand contains instructions to find all tasks that match a
+ * specific pattern.
+ */
+public class FindCommand extends Command {
+
+    /**
+     * Creates a new FindCommand that can search for the given details.
+     *
+     * @param details   String containing details that found tasks should
+     *                  contain
+     */
+    public FindCommand(String details) {
+        super(details);
+    }
+
+    /**
+     * Displays all tasks that match the details contained in the current
+     * FindCommand. Tasks are evaluated using their default toString() values
+     * and are matched against the current command's details using the String
+     * method contains(String).
+     *
+     * @param tasks             List of tasks
+     * @param ui                User interface
+     * @param storage           Hard disk storage
+     */
+    @Override
+    public void execute(TaskList tasks, TextUi ui, Storage storage) {
+        TaskList foundTasks = new TaskList();
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            if (task.toString().contains(this.details)) {
+                foundTasks.add(task);
+            }
+        }
+        ui.showList(foundTasks);
```

Comment:
Would it be more ideal to print the found tasks with their original list numbers? Then there would be no confusion if the user wanted to perform commands on the found tasks.